src/main.rs

```diff
@@ -253,6 +274,16 @@ fn main() -> Result<i32> {
         if is_redirect && exit_code != 0 {
             eprintln!("\n{}: warning: HTTP {}\n", env!("CARGO_PKG_NAME"), status);
         }
+
+        if let Some(ref mut s) = session {
+            if orig_url.host() == response.url().host() {
```

Comment:
This is not quite enough. We also need to check the scheme and the port. [`remove_sensitive_headers`](https://github.com/seanmonstar/reqwest/blob/18dfac4fe2590f2e3a27e808c72fb26e6d9af98b/src/redirect.rs#L234) is a good example.

HTTPie makes `requests`'s session handling do the hard work. That even handles the authorization header. But I don't think that exists for `reqwest`.

While digging this up I did some more thinking about cookies. Ideally we would capture cookies that were set in intermediate responses, even if the final response is from another host. 

If we enable the `cookies` feature we can use [`cookie_provider()`](https://docs.rs/reqwest/0.11.3/reqwest/blocking/struct.ClientBuilder.html#method.cookie_provider) to set a cookie jar. It doesn't have a way to get the jar out, but it takes an `Arc` so we can keep using it after passing it into the ClientBuilder.

I assume it does safe domain handling. I think that's the sanest option for now. (Unfortunately I think it'd require rewriting some of the other code in this PR.)

Reply:
>HTTPie makes requests's session handling do the hard work. That even handles the authorization header. But I don't think that exists for reqwest.

I think reqwest does remove sensitive headers on cross-domain redirect. See https://github.com/seanmonstar/reqwest/blob/c6eb2c4fcbc3934b119772e0997d91671ff68dc7/src/redirect.rs#L234